### PR TITLE
Refactors most things that check if the player is walking to check actual speed (and a few other things)

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
adds the carefulmovement() proc. This proc, provided a mob
-is conscious
-is not buckled to something 
-has a speed equal to or less than the speed of a walking mob
will return true.  this replaces most checks that check if the mob is walking

## Why It's Good For The Game
walking is a rudimentary check that can be bypassed by speedups, vehicles, etc. A more standardized approach is both more difficult to bypass and easier to modify.

in the future, i may 
-make walking not stalk with other slowdowns
-make carefulmove() return true when standing still for a set amount of time

## Changelog
:cl:
code: most instances of code that check if the player is walking now check speed. This should have very little difference gameplay-wise, though vehicles will now no longer allow you to count as walking
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon_movement.dm [hunk 1]
```diff
@@ -35,3 +35,12 @@
 			adjust_nutrition(-(HUNGER_FACTOR/10))
 			if(m_intent == MOVE_INTENT_RUN)
 				adjust_nutrition(-(HUNGER_FACTOR/10))
+	
+/mob/living/carbon/carefulmovement()
```
Comment (axpov) [inline code/modules/mob/living/carbon/carbon_movement.dm]: "careful movement" sounds more like an action than a check. Should probably be renamed to make it obvious it's checking if the user is moving carefully

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 2]
```diff
@@ -79,3 +79,12 @@
 	if(dna.species.space_move(src))
 		return TRUE
 	return ..()
+
+/mob/living/carbon/human/carefulmovement()
```
Comment (axpov) [inline code/modules/mob/living/carbon/human/human_movement.dm]: Some of these checks are redundant with the parent and this proc should probably just extend that

Comment (zeskorion) [reply code/modules/mob/living/carbon/human/human_movement.dm]: parent also checks move intent while this needs to check species to check exact speed. this aint feasible

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 3]
```diff
@@ -79,3 +79,12 @@
 	if(dna.species.space_move(src))
 		return TRUE
 	return ..()
+
+/mob/living/carbon/human/movingcarefully()
+	if(buckled) //sitting down, being in a vehicle, etc
+		return FALSE 
+	if(stat) //iron out edgecases
+		return FALSE
+	if((cached_multiplicative_slowdown + dna.species.movement_delay(src)) >= CONFIG_GET(number/movedelay/walk_delay) && mobility_flags & MOBILITY_STAND) //later, i want walk to not slow you down past walk delay. later.
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/human_movement.dm:88]: Species that move slower shouldn't be counted as walking if they are on their species run intent. You should check if cached_multiplicative_slowdown > 0 (or whatever the default is) and leave the config out of it

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human_movement.dm (hunk 3 above)
Comment (zeskorion) [reply code/modules/mob/living/carbon/human/human_movement.dm:88]: the intent here is not to limit to walking, but to nerf speed in combat. If a character is moving slow enough to consistently click on, it's fine.

anyways, reopen this pr, please.

**Pull request closed without merging**